back-end/app.js

```diff
@@ -61,14 +60,41 @@ app.get("/", (req, res) => {
   })
 })
 
-// const Job = require("./models/jobDetails")
 
-// app.post("/jobs/new", async (req, res) => {
-//   const newJob = await Job.create(req.body)
+// *** jobDetails - CRUD Operations ***
+const Job = require("./models/jobDetails")
+// jobDetails: Create
+app.post("/jobs/new", async (req, res) => {
+  const newJob = await Job.create(req.body)
+
+  res.json(newJob)
+})
+// jobDetails: Read
+app.get("/jobs/:id", async (req, res) => {
+  const foundJob = await Job.findOne({"_id": req.params.id})
+
+  res.json(foundJob)
+})
+// jobDetails: Update (the status of the job)
+app.put("/jobs/:id", async (req, res) => {
+  console.log(req.body.changeStatus)
+  console.log(req.params.id)
+
+  await Job.findOneAndUpdate(
+    {"_id": req.params.id},
+    {status: req.body.changeStatus},
+    {returnOriginal: false}
+  )
+})
```

Comment:
I believe the reason why there is a slight delay with the update command is because there isn't a `res.json` command here. So whenever the frontend sends over a request to update the job, it will expect a response back (once it receives the response back, it will then update the text on the frontend), but since there is no `res.json`, a response is never sent back and the frontend is stuck waiting for the response. So you can put something like `res.json("Successfully updated")` or maybe just send back the updated job, it probably shouldn't matter too much! :)